### ISSUE TITLE
Rustdoc does not warn about unknown, renamed, or removed lints

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#75884]
Found in https://github.com/tokio-rs/tracing/issues/940#issuecomment-679249098.

```
$ cat broken.rs 
#![deny(x)]
$ rustdoc broken.rs  # no output
$ rustc broken.rs 
warning: unknown lint: `x`
 --> broken.rs:1:9
  |
1 | #![deny(x)]
  |         ^
  |
  = note: `#[warn(unknown_lints)]` on by default
$ rustdoc --version
rustdoc 1.45.2 (d3fb005a3 2020-07-31)
```

**Comment by jyn514** [comment]
`renamed_and_unknown_lints` is also not running, which is even worse in this case because it means no one will see the message about intra-doc links being renamed.
```
$ rustc +nightly broken.rs  --crate-type lib
warning: lint `intra_doc_link_resolution_failure` has been renamed to `broken_intra_doc_links`
 --> broken.rs:1:9
  |
1 | #![deny(intra_doc_link_resolution_failure)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use the new name: `broken_intra_doc_links`
  |
  = note: `#[warn(renamed_and_removed_lints)]` on by default
$ rustdoc +nightly broken.rs  # no output
```

**Comment by jyn514** [comment]
Relevant code; https://github.com/rust-lang/rust/blob/6e3e74dd719334e84edd70ba0a842f5d4f94d266/src/librustdoc/core.rs#L225-L228
